src/EnrollWebauthnController.js

```diff
@@ -0,0 +1,157 @@
+          var deferred = Q.defer();
+
+          var options = activation;
+
+          options.challenge = CryptoUtil.strToBin(activation.challenge);
+          options.user.id = CryptoUtil.strToBin(activation.user.id);
+
+          navigator.credentials.create({
+            publicKey: options
+          }).then(function (newCredential) {
+            deferred.resolve(transaction.activate({
+              attestation: CryptoUtil.binToStr(newCredential.response.attestationObject),
+              clientData: CryptoUtil.binToStr(newCredential.response.clientDataJSON)
+            }));
+          }).catch(function (err) {
+            deferred.reject({
+              xhr: {responseJSON: {errorSummary: err}}
+            });
+          });
+
+          return deferred.promise;
+        });
+      }
+    },
+
+    Form: {
+      title: _.partial(Okta.loc, 'enroll.u2f.title', 'login'),
+      save: _.partial(Okta.loc, 'enroll.u2f.save', 'login'),
+      noCancelButton: true,
+      hasSavingState: false,
+      autoSave: true,
+      className: 'enroll-u2f-form',
+      noButtonBar: function () {
+        return navigator.credentials.create == null;
+      },
+      modelEvents: {
+        'request': '_startEnrollment',
+        'error': '_stopEnrollment'
+      },
+      formChildren: function () {
+        var result = [];
+
+        // TODO: handle non-webauthn browser flow
+        if (navigator.credentials.create == null) {
+          var errorMessageKey = 'u2f.error.factorNotSupported';
+          if (this.options.appState.get('factors').length === 1) {
+            errorMessageKey = 'u2f.error.factorNotSupported.oneFactor';
+          }
+          result.push(FormType.View(
+            {View: new HtmlErrorMessageView({message: Okta.loc(errorMessageKey, 'login')})},
+            {selector: '.o-form-error-container'}
+          ));
+        }
+        else {
+          //There is html in enroll.u2f.general2 in our properties file, reason why is unescaped
+          result.push(FormType.View({
+            View: '<div class="u2f-instructions"><ol>\
+          <li>{{{i18n code="enroll.u2f.general2" bundle="login"}}}</li>\
+          <li>{{i18n code="enroll.u2f.general3" bundle="login"}}</li>\
+          </ol></div>'
+          }));
+
+          result.push(FormType.View({
+            View: '\
+          <div class="u2f-enroll-text hide">\
+            <p>{{i18n code="enroll.u2f.instructions" bundle="login"}}</p>\
+            <p>{{i18n code="enroll.u2f.instructionsBluetooth" bundle="login"}}</p>\
+            <div data-se="u2f-devices" class="u2f-devices-images">\
+              <div class="u2f-usb"></div>\
+              <div class="u2f-bluetooth"></div>\
+            </div>\
+            <div data-se="u2f-waiting" class="okta-waiting-spinner"></div>\
+          </div>'
+          }));
+        }
+
+        return result;
+      },
+
+      _startEnrollment: function () {
+        this.$('.u2f-instructions').addClass('hide');
```

Comment:
Is there a reason you're using the class hide for these two, and using the method hide for o-form-button-bar?

And in general, since we're triggering these ourselves, could just squash this down to one `toggle` function and use the [toggleClass](http://api.jquery.com/toggleclass/) method with params for toggling true or false.